py3status/modules/speedtest.py

```diff
@@ -0,0 +1,299 @@
+                    )
+
+                    # create "quality" #maybe bad name
+                    if previous_data and "total" in previous_data:
+                        if current_data["total"] >= previous_data["total"]:
+                            quality = "faster"
+                        else:
+                            quality = "slower"
+                    else:
+                        if current_data["total"] <= 0:
+                            quality = "bad"
+                        else:
+                            quality = "ok"
+                    current_data["quality"] = quality
+
+            # zero-ing if not fetched, raw version and units convertion
+            for x in ["download", "upload", "bytes_received", "bytes_sent"]:
+                unit = self.unit_size if 'bytes' in x else self.unit_bitrate
+                current_data[x] = current_data.get(x, 0)
+                current_data[x + "_raw"] = current_data[x]
+                current_data[x], current_data[x + "_unit"] = self.py3.format_units(
+                    current_data[x], unit=unit, si=self.si_units
+                )
+
+            # extra data, not sure we want to expose
+            for x in ["server", "client"]:
+                if x in current_data:
+                    for y in current_data[x]:
+                        current_data[x + "_" + y] = current_data[x][y]
+                    del current_data[x]
+
+            # store last data fetched
+            self.py3.storage_set("speedtest_data", current_data)
+
+            # get speedtest result url
+            if current_data and "share" in current_data:
+                self.url = current_data["share"]
+
+            # create placeholders
+            speedtest_data.update(current_data)
+            if previous_data:
+                speedtest_data.update(
+                    {"previous_" + k: v for (k, v) in previous_data.items()}
+                )
+
+            # # cast
+            speedtest_data.update(
+                {
+                    k: float(v)
+                    for (k, v) in speedtest_data.items()
+                    if isinstance(v, Number)
+                }
+            )
+
+            # thresholds
+            for x in self.thresholds:
+                if x in speedtest_data:
+                    self.py3.threshold_get_color(speedtest_data[x], x)
+
+        return {
+            "cached_until": cached_until,
+            "full_text": self.py3.safe_format(self.format, speedtest_data),
+        }
+
+    def on_click(self, event):
+        button = event["button"]
+        if button == self.button_share and self.url:
+            self.py3.command_run("xdg-open %s" % self.url)
+        elif button != self.button_refresh:
+            self.py3.prevent_refresh()
+
+
+if __name__ == "__main__":
+    """
+    Run module in test mode.
+    """
+    from py3status.module_test import module_test
+
+    module_test(Py3status)
```

Comment:
There is a Python API too. https://github.com/sivel/speedtest-cli/wiki
Whether it's better or not, I dunno. I just wanted to mention this.